boot/modeenv.go

```diff
@@ -195,3 +196,40 @@ func splitModeenvStringList(v string) []string {
 func asModeenvStringList(v []string) string {
 	return strings.Join(v, ",")
 }
+
+// deepEqual compares two modeenvs to ensure they are textually the same. It
+// does not consider whether the modeenvs were read from disk or created purely
+// in memory. It also does not sort or otherwise mutate any sub-objects,
+// performing simple strict verification of sub-objects.
+func (m *Modeenv) deepEqual(m2 *Modeenv) bool {
```

Comment:
Shouldn't the name be deepEquals ("s")?

Reply:
no, as per @pedronis go convention is `deepEqual`

Reply:
Ah, ok, that's fine.